src/main/java/com/project/stockexchangeappbackend/service/StockServiceImpl.java

```diff
@@ -10,34 +11,67 @@
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
 
+import javax.persistence.EntityExistsException;
 import javax.persistence.EntityNotFoundException;
+import java.util.Optional;
 
 @Service
 @RequiredArgsConstructor
 public class StockServiceImpl implements StockService {
 
-    private final StockRepository repository;
+    private final StockRepository stockRepository;
 
     @Override
     @LogicBusinessMeasureTime
     @Transactional(readOnly = true)
     public Stock getStockById(Long id) {
-        return repository.findById(id).orElseThrow(() -> new EntityNotFoundException("Stock Not Found"));
+        return stockRepository.findById(id).orElseThrow(() -> new EntityNotFoundException("Stock Not Found"));
     }
 
     @Override
     @LogicBusinessMeasureTime
     @Transactional(readOnly = true)
     public Page<Stock> getStocks(Pageable pageable, Specification<Stock> specification) {
-        return repository.findAll(specification, pageable);
+        return stockRepository.findAll(specification, pageable);
     }
 
     @Override
     @LogicBusinessMeasureTime
     public Stock getStockByAbbreviation(String abbreviation) {
-        return repository.findByAbbreviationIgnoreCase(abbreviation).orElseThrow(() ->
+        return stockRepository.findByAbbreviationIgnoreCase(abbreviation).orElseThrow(() ->
                 new EntityNotFoundException("Stock Not Found"));
     }
 
+    @Override
+    @LogicBusinessMeasureTime
+    @Transactional
+    public void updateStock(StockDTO stockDTO, String id) {
+        Optional<Stock> stockOptional = stockRepository.findByNameOrAbbreviationIgnoreCase(stockDTO.getName().trim(),
+                stockDTO.getAbbreviation().trim());
+        if (stockOptional.isPresent()) {
+            try {
+                if (!stockOptional.get().getId().equals(new Long(id))) {
+                    throw new EntityExistsException("Stock with given details already exists");
+                }
+            } catch (NumberFormatException e) {
+                if (!stockOptional.get().getAbbreviation().equals(id)) {
+                    throw new EntityExistsException("Stock with given details already exists");
+                }
+            }
```

Comment:
remove this block of code. And check it after when you retrieve exact Stock. In line 64 put:
if (stockOptional.isPresent() && !stock.getId().equal(stockOptional.get().getId()) {
...